JS.md

```diff
@@ -13,6 +13,22 @@ basically equivalent to
 `new WebAssembly.Instance(new WebAssembly.Module(bytes), imports)`
 as defined below and will be removed at some point in the future.*
 
+## Traps
+
+Whenever WebAssembly semantics specify a [trap](AstSemantics.md#traps),
+a `WebAssembly.RuntimeError` object is thrown. WebAssembly code (currently)
+has no way to catch this exception and thus the exception will necessarily
+propagate to the innermost non-WebAssembly caller (either the browser or
+JavaScript) where it is handled like a normal JavaScript exception.
+
+If WebAssembly calls JavaScript via import and the JavaScript throws an
+exception, the exception is propagated through the WebAssembly activation to the
+enclosing caller.
+
+Because JavaScript exceptions can be handled, and JavaScript can continue to
+call WebAssembly exports after a trap has been handled, traps do not, in
+general, prevent future execution.
```

Comment:
Right, that's vaguely what I recalled. Should you then mention something about observability of trap ordering? I think that's what we'd concluded (e.g. two accesses which would trap are observed to trap in program order). This seems like the right place to do so, but I'm happy with any other place you find more appropriate.